src/core/mesh/qgsmeshdataprovider.h

```diff
@@ -290,7 +290,7 @@ class CORE_EXPORT QgsMeshDatasetSourceInterface SIP_ABSTRACT
      *
      * On success, the mesh's dataset group count is changed
      *
-     * \param path destination path of the stored file in form DRIVER_NAME:path
+     * \param path destination path of the stored file in form DRIVER_NAME:"path"
```

Comment:
This is effectively API change, since the previous format DRIVER_NAME:path (e.g. on Linux where it worked) will not be parsed anymore. 

I suggest
1. deprecate `persistDatasetGroup`, add not that not working on windows and keep the logic on master to split DRIVER_NAME https://github.com/qgis/QGIS/blob/2cdd2eea5763fe8e373ba1cfad4a4d3a13fee621/src/analysis/mesh/qgsmeshcalculator.h#L163
2. create new `persistDatasetGroup` where you add new parameter driver (explicitely) similar to this https://github.com/qgis/QGIS/blob/2cdd2eea5763fe8e373ba1cfad4a4d3a13fee621/src/analysis/mesh/qgsmeshcalculator.h#L140 and deprecated function 1) will just split by driver_name and call 2). Use in QgsMeshCalculator new function.